src/main/java/com/minecolonies/coremod/network/messages/HutRenameMessage.java

```diff
@@ -0,0 +1,110 @@
+package com.minecolonies.coremod.network.messages;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.permissions.Action;
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Message to execute the renaiming of the townHall.
+ */
+public class HutRenameMessage extends AbstractMessage<HutRenameMessage, IMessage>
+{
+    /**
+     * The max length of the name.
+     */
+    private static final int MAX_NAME_LENGTH = 25;
```

Comment:
this should be also errored when checking correctness of the input name in the window

Reply:
Here I agree